File: 55. Jump Game/55. Jump Game.cs
public class Solution {
    public bool CanJump(int[] nums) {
        int maxReach = 0;
        
        for (int i = 0; i <= maxReach && i < nums.Length; i++) {
            maxReach = Math.Max(maxReach, i + nums[i]);
            if (maxReach >= nums.Length - 1) {
                return true;
            }
        }
        
        return false;
    }
}

File: 55. Jump Game/55. Jump Game.dart
class Solution {
  bool canJump(List<int> nums) {
    int maxReach = 0;
    
    for (int i = 0; i <= maxReach && i < nums.length; i++) {
      maxReach = maxReach < i + nums[i] ? i + nums[i] : maxReach;
      if (maxReach >= nums.length - 1) {
        return true;
      }
    }
    
    return false;
  }
}